kong/clustering/services/sync/rpc.lua

```diff
@@ -0,0 +1,324 @@
+local _M = {}
+local _MT = { __index = _M, }
+
+
+local txn = require("resty.lmdb.transaction")
+local declarative = require("kong.db.declarative")
+local constants = require("kong.constants")
+local concurrency = require("kong.concurrency")
+
+
+local insert_entity_for_txn = declarative.insert_entity_for_txn
+local delete_entity_for_txn = declarative.delete_entity_for_txn
+local DECLARATIVE_HASH_KEY = constants.DECLARATIVE_HASH_KEY
+local CLUSTERING_SYNC_STATUS = constants.CLUSTERING_SYNC_STATUS
+local SYNC_MUTEX_OPTS = { name = "get_delta", timeout = 0, }
+
+
+local pairs = pairs
+local ipairs = ipairs
+local fmt = string.format
+local ngx_null = ngx.null
+local ngx_log = ngx.log
+local ngx_ERR = ngx.ERR
+local ngx_INFO = ngx.INFO
+local ngx_DEBUG = ngx.DEBUG
+
+
+-- number of versions behind before a full sync is forced
+local FULL_SYNC_THRESHOLD = 512
+
+
+function _M.new(strategy)
+  local self = {
+    strategy = strategy,
+  }
+
+  return setmetatable(self, _MT)
+end
+
+
+function _M:init_cp(manager)
+  -- CP
+  -- Method: kong.sync.v2.get_delta
+  -- Params: versions: list of current versions of the database
+  -- { { namespace = "default", current_version = 1000, }, }
+  local purge_delay = manager.conf.cluster_data_plane_purge_delay
+
+  local function gen_delta_result(res, wipe)
+    return { default = { deltas = res, wipe = wipe, }, }
+  end
+
+  manager.callbacks:register("kong.sync.v2.get_delta", function(node_id, current_versions)
+    ngx_log(ngx_DEBUG, "[kong.sync.v2] config push (connected client)")
+
+    local rpc_peers
+    if kong.rpc then
+      rpc_peers = kong.rpc:get_peers()
+    end
+
+    local default_namespace
+    for namespace, v in pairs(current_versions) do
+      if namespace == "default" then
+        default_namespace = v
```

Comment:
```suggestion
        default_namespace = v
        break
```

Need to consider whether to break here.